src/pytorch_lightning/plugins/io/checkpoint_plugin.py

```diff
@@ -52,6 +52,10 @@ def load_checkpoint(self, path: _PATH, map_location: Optional[Any] = None) -> Di
                 locations.
 
         Returns: The loaded checkpoint.
+
+        Note:
+            MAP_LOCATION = Optional[Union[Callable, _DEVICE, Callable[[_DEVICE], _DEVICE], Dict[_DEVICE, _DEVICE]]]
+            _DEVICE = Union[torch.device, str, int]
```

Comment:
This is not needed. The types get expanded anyway. Let's remove it

Reply:
reverted to https://github.com/Lightning-AI/lightning/pull/13932/commits/0502a4e9fe2f1b229265de8ae82a129e704f65ee and synced with upstream in https://github.com/Lightning-AI/lightning/pull/13932/commits/82fc4528431fd3412e30782ba64bcc65d2adba38